### PR TITLE
Fix some bugs in print_summary

### PR DESCRIPTION
**Pull request by bburke38**
- Add a catch for `CompositeFunction` in `_print_functions()` since the composite functions don't have the same properties as a typical function.
- Give composite functions an `analysis_type` property
- When verifying vartype, print which vartype was not recognized.

### CODE REVIEW

File: funtofem/model/_base.py [hunk 1]
```diff
@@ -199,9 +199,10 @@ def verify_vartype(self, vartype):
             "rigid_motion",
             "shape",
             "controls",
+            "thermal",
         ]:
             print(
-                "Warning: vartype specified is not a recognized variable type",
+                f'Warning: vartype "{vartype}" is not a recognized variable type',
```
Comment (sean-engelstad) [inline funtofem/model/_base.py:205]: Can you make an attribute VAR_TYPES = ["structural", "aerodynamic", "rigid_motion", "shape", "controls", "thermal"] above the __init__ constructor. This way people can easily see the available types at the top of the file. Then you can do:

```
if not vartype in self.VAR_TYPES:
    print(f"Warning ... etc")
```

**Review by sean-engelstad — CHANGES_REQUESTED** [review]
Small request: add VAR_TYPES list to the top of the class that way people can see the available variable analysis types easier instead of having to read that far down into the class.

File: funtofem/model/_base.py (hunk 1 above)
Comment (bburke38) [reply funtofem/model/_base.py:207]: Good idea, I implemented it in the latest commit

**Review by sean-engelstad — APPROVED** [review]
Approved

**Pull request merged**